USE_SHM_MEM compile flag is broken

Not sure whether enabling `USE_SHM_MEM` is to blame, or the fact that `PKG_MALLOC` is disabled. 

Regardless, among lots of warnings:

```
Compiling mi/mi.c
mi/mi.c: In function ‘register_mi_cmd’:
mi/mi.c:142:17: warning: implicit declaration of function ‘pkg_realloc’ [-Wimplicit-function-declaration]
  cmds = (struct mi_cmd*)pkg_realloc( mi_cmds,
                 ^
mi/mi.c:142:9: warning: cast to pointer from integer of different size [-Wint-to-pointer-cast]
  cmds = (struct mi_cmd*)pkg_realloc( mi_cmds,
```

there is also an error:

```
Compiling cfg.tab.c
cfg.y: In function ‘yyparse’:
cfg.y:765:4: error: ‘event_pkg_threshold’ undeclared (first use in this function)
    event_pkg_threshold=$3;
    ^
cfg.y:765:4: note: each undeclared identifier is reported only once for each function it appears in
```

TODO: assess whether `USE_SHM_MEM` is still useful in the first place, then fix appropriately.

--- a/cfg.y
+++ b/cfg.y
@@ -759,10 +759,16 @@ assign_stm: DEBUG EQUAL snumber
 		| EVENT_SHM_THRESHOLD EQUAL error { yyerror("int value expected"); }
 		| EVENT_PKG_THRESHOLD EQUAL NUMBER {
 			#ifdef STATISTICS
+			#ifdef USE_SHM_MEM
+				warn("No PKG memory, all allocations are mapped to SHM; "
+					"Use event_shm_threshold instead or recompile with PKG_MALLOC "
+					"instead of USE_SHM_MEM in order to have separate PKG memory");
+			#else
 			if ($3 < 0 || $3 > 100)
 				yyerror("PKG threshold has to be a percentage between "
 					"0 and 100");
 			event_pkg_threshold=$3;
+			#endif
 			#else
 			yyerror("statistics support not compiled in");
 			#endif

--- a/main.c
+++ b/main.c
@@ -758,8 +758,36 @@ int main(int argc, char** argv)
 					};
 
 					break;
+			case 'm':
+					shm_mem_size=strtol(optarg, &tmp, 10) * 1024 * 1024;
+					if (tmp &&(*tmp)){
+						LM_ERR("bad shmem size number: -m %s\n", optarg);
+						goto error00;
+					};
+					break;
+			case 'u':
+					user=optarg;
+					break;
+			case 'g':
+					group=optarg;
+					break;
 		}
 	}
+
+	/* get uid/gid */
+	if (user){
+		if (user2uid(&uid, &gid, user)<0){
+			LM_ERR("bad user name/uid number: -u %s\n", user);
+			goto error00;
+		}
+	}
+	if (group){
+		if (group2gid(&gid, group)<0){
+			LM_ERR("bad group name/gid number: -u %s\n", group);
+			goto error00;
+		}
+	}
+
 	/*init pkg mallocs (before parsing cfg but after parsing cmd line !)*/
 	if (init_pkg_mallocs()==-1)
 		goto error00;
@@ -783,11 +811,7 @@ int main(int argc, char** argv)
 					cfg_log_stderr=1; /* force stderr logging */
 					break;
 			case 'm':
-					shm_mem_size=strtol(optarg, &tmp, 10) * 1024 * 1024;
-					if (tmp &&(*tmp)){
-						LM_ERR("bad shmem size number: -m %s\n", optarg);
-						goto error00;
-					};
+					/* ignoring it, parsed previously */
 					break;
 			case 'M':
 					/* ignoring it, parsed previously */
@@ -876,10 +900,10 @@ int main(int argc, char** argv)
 					chroot_dir=optarg;
 					break;
 			case 'u':
-					user=optarg;
+					/* ignoring it, parsed previously */
 					break;
 			case 'g':
-					group=optarg;
+					/* ignoring it, parsed previously */
 					break;
 			case 'P':
 					pid_file=optarg;
@@ -955,21 +979,7 @@ try_again:
 	yydebug = 1;
 #endif
 
-		/* get uid/gid */
-	if (user){
-		if (user2uid(&uid, &gid, user)<0){
-			LM_ERR("bad user name/uid number: -u %s\n", user);
-			goto error00;
-		}
-	}
-	if (group){
-		if (group2gid(&gid, group)<0){
-			LM_ERR("bad group name/gid number: -u %s\n", group);
-			goto error00;
-		}
-	}
-
-		/*init shm mallocs
+	/*  init shm mallocs
 	 *  this must be here
 	 *     -to allow setting shm mem size from the command line
 	 *     -it must be also before init_timer and init_tcp

--- a/mem/mem.c
+++ b/mem/mem.c
@@ -77,8 +77,17 @@ int init_pkg_mallocs(void)
 			pkg_mem_size );
 		return -1;
 	}
-#endif
+#elif defined USE_SHM_MEM
+	if (shm_mem_init()<0) {
+		LM_CRIT("could not initialize shared memory pool, exiting...\n");
+		 fprintf(stderr, "Too much shared memory demanded: %ld\n",
+			shm_mem_size );
+		return -1;
+	}
 	return 0;
+#else
+	return 0;
+#endif
 }
 
 
@@ -190,12 +199,15 @@ void pkg_threshold_check(void)
 
 int init_shm_mallocs(void)
 {
+	#ifndef USE_SHM_MEM
 	if (shm_mem_init()<0) {
 		LM_CRIT("could not initialize shared memory pool, exiting...\n");
 		 fprintf(stderr, "Too much shared memory demanded: %ld\n",
 			shm_mem_size );
 		return -1;
 	}
+	#endif
+
 	return 0;
 }
 

--- a/mem/mem.h
+++ b/mem/mem.h
@@ -186,6 +186,7 @@ void set_pkg_stats(pkg_status_holder*);
 #elif defined(USE_SHM_MEM)
 #	include "shm_mem.h"
 #	define pkg_malloc(s) shm_malloc((s))
+#	define pkg_realloc(p, s) shm_realloc((p), (s))
 #	define pkg_free(p)   shm_free((p))
 #	define pkg_status()  shm_status()
 #	define MY_PKG_GET_SIZE()

